src/sleep/sleep.c

```diff
@@ -211,6 +211,12 @@ static int execute(char **modes, char **states) {
                         return log_error_errno(r, "Failed to write mode to /sys/power/disk: %m");;
         }
 
+        char action_env[STRLEN("SYSTEMD_SLEEP_ACTION=") + strlen(action)];
```

Comment:
uh I'm not sure we want varlen arrays - Lennart will know - given the names are fixed, just strjoina("SYSTEMD_SLEEP_ACTION=", action) would be better maybe?

Reply:
Thanks.